python/hidet/runtime/compiled_task.py

```diff
@@ -176,8 +176,21 @@ def run_async(self, inputs):
             _check_inputs(self.meta_data.inputs, inputs)
 
         outputs = self.create_outputs()
+
+        has_vcuda = len(inputs) > 0 and all(inp.device.is_vcuda() for inp in inputs)
+        if has_vcuda:
+            for inp in inputs:
+                inp.move_from_vcuda()
+
         candidate = self.candidates[self.pick_best_candidate(inputs, outputs)]
         candidate(*inputs, *outputs)
+
+        if has_vcuda:
+            for inp in inputs:
+                inp.move_to_vcuda()
+            for outp in outputs:
+                outp.move_to_vcuda()
+
```

Comment:
There are two ways to implement "vcuda" device:
1. [The current version]: we copy the tensor to cuda before we run the operator and copy back to cpu when we finish it.
2. We can also reply on the unified addressing of cuda. (see [here](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#unified-virtual-address-space) and [here](https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__UNIFIED.html)). Essentially, the unified addressing allows the cuda kernel access the host memory (page-locked) directly. Thus, we do not need to copy to and from the cuda device memory.

I prefer the second way. With it, we do not need to update our `run_async` function.

Reply:
this is a great suggestion and it works! thank you

Reply:
One minor issue is that the cpu tensor allocated by other library (e.g., pytorch and numpy) will not use `cudaMallocHost` to allocate, and it is not page-locked. Thus, when we convert the tensors from them, we should allocate the tensor by hidet instead of sharing the weights (on the platform that cuda is avaliable).

Reply:
I think this case is fine because if users give a cpu tensor from other library and expect it to work on CUDA device that is already wrong

Reply:
runtime check should catch this

Reply:
Make sense.